evm_arithmetization/src/cpu/kernel/asm/core/access_lists.asm

```diff
@@ -12,6 +12,13 @@
 // Initialize SEGMENT_ACCESSED_ADDRESSES
 global init_access_lists:
     // stack: (empty)
+
+    // Reset access lists data.
+    PUSH 0 %mstore_global_metadata(@GLOBAL_METADATA_ACCESS_LIST_DATA_COST)
+    PUSH 0 %mstore_global_metadata(@GLOBAL_METADATA_ACCESS_LIST_RLP_LEN)
+    PUSH 0 %mstore_global_metadata(@GLOBAL_METADATA_ACCESS_LIST_RLP_START)
+    PUSH 0 %mstore_global_metadata(@GLOBAL_METADATA_ACCESSED_STORAGE_KEYS_LEN)
```

Comment:
```suggestion
```
`@GLOBAL_METADATA_ACCESSED_STORAGE_KEYS_LEN` is already updated to the right value (`@SEGMENT_ACCESSED_STORAGE_KEYS + 4` ) on line 52 (similarly for `@GLOBAL_METADATA_ACCESSED_ADDRESSES_LEN` )

Reply:
But don't they need to be reset to 0 after each transaction?

Reply:
No because we initialize the empty lists with `U256::MAX`. Alonso is right, we're updating these two values just below, effectively overwriting the (erroneous) zero.